python/cudf/cudf/_lib/io/CMakeLists.txt

```diff
@@ -20,7 +20,16 @@ rapids_cython_create_modules(
   LINKED_LIBRARIES "${linked_libraries}" MODULE_PREFIX io_ ASSOCIATED_TARGETS cudf
 )
 
-set(targets_using_numpy io_datasource io_utils)
-foreach(target IN LISTS targets_using_numpy)
+find_package(Python 3.8 REQUIRED COMPONENTS Interpreter)
```

Comment:
@robertmaynard this is more of a CMake best practices question; should we duplicate this `find_package` here so that we don't rely on the global state of the find happening in a higher-level CMakeLists.txt file, or should we avoid duplicating and trust future devs (most like you or me when we upgrade to CMake 3.25) to make the necessary adjustments?

Reply:
I think in this given situtation where we are building python bindings it is reasonable to expect that python has been found at the root most level.

Reply:
Looking at this more. We should hoist up the `execute_process` to determine the pyarrow header location so we aren't doing that 2 times for each configuration.

Reply:
> We should hoist up the `execute_process` to determine the pyarrow header location so we aren't doing that 2 times for each configuration.

Not sure how trivial this is, should we take this up in a follow-up PR?

Reply:
It's straightforward, a deletion of the second instance is sufficient since they're already nested directories. I will make the change, test locally, and push.

Reply:
This is done.